core/src/main/java/org/apache/gravitino/listener/api/event/AlterTablePreEvent.java

```diff
@@ -30,7 +30,7 @@ public class AlterTablePreEvent extends TablePreEvent {
 
   public AlterTablePreEvent(String user, NameIdentifier identifier, TableChange[] tableChanges) {
     super(user, identifier);
-    this.tableChanges = tableChanges;
+    this.tableChanges = tableChanges.clone();
```

Comment:
In post-event, the objects are cloned to prevent the user from changing the object. But in pre-event, I prefer to relax the constraint to allow users to make changes to in inject some custom logic.